src/Conference/ConferenceDetails.test.js

```diff
@@ -41,30 +46,40 @@ afterEach(() => {
 
 describe("ConferenceDetails", () => {
   const mockData = {
-    data: {
-      id: 1,
-      name: "Festival of Marketing",
-      dateTime: "2019-11-12T12:34:11Z",
-      city: "London",
-      description:
-        "From Festivalofmarketing.com: The Festival of Marketing is a unique experience where ambitious marketers can discover, learn, celebrate and shape the future together. As the largest global event dedicated to brand marketers, the Festival reflects the very nature of ...",
-      topic: "Marketing"
-    }
+    id: 1,
+    name: "Festival of Marketing",
+    dateTime: "2019-11-12T12:34:11Z",
+    city: "London",
+    description:
+      "From Festivalofmarketing.com: The Festival of Marketing is a unique experience where ambitious marketers can discover, learn, celebrate and shape the future together. As the largest global event dedicated to brand marketers, the Festival reflects the very nature of ...",
+    topic: "Marketing"
   };
 
-  it("renders conference details", done => {
-    expect.assertions(1);
+  it("renders conference details", async () => {
+    expect.assertions(6);
 
-    axios.get.mockResolvedValue(mockData);
+    const history = createMemoryHistory();
+    history.push("/1");
 
-    useParams.mockReturnValue({ id: "1" });
+    const apiReturnValue = Promise.resolve(mockData);
+    getConferenceById.mockImplementation(() => apiReturnValue);
 
-    const component = mount(<ConferenceDetails />);
+    let wrapper;
+    await act(async () => {
+      wrapper = mount(
+        <Router history={history}>
+          <ConferenceDetails />
+        </Router>
+      );
 
-    setImmediate(() => {
-      component.update();
-      expect(component).toMatchSnapshot();
-      done();
+      await apiReturnValue;
     });
+    wrapper.update();
+    expect(wrapper.find("h3.name").length).toBe(1);
```

Comment:
You dont need `h3` in this find. Its a bit verbose but it will still pass

Reply:
Hi Jack, when we remove the h3 the test fails saying received 2 instead of 1. We aren't sure why this is happening

Reply:
Interesting, it's finding another element (not a h3) with classname 'name'. Do you have this somewhere else?

Reply:
The only place we reference 'name' is in conference, we don't use 'className = name' anywhere else as far as we can see

Reply:
should be introducing data-test tags for specificity?

Reply:
@longstaff-elsevier we could do that, i feel like a specific className is enough at this point or alternatively if the hmtl element is unique you can import the styled component and find it